executor/include/executor/default/cuda_executor.hpp

```diff
@@ -35,9 +35,21 @@ struct is_executor_available<cuda_executor> : std::true_type {};
 
 template <typename Blocking = blocking_t::always_t,
           typename ProtoAllocator = std::allocator<void>>
-struct cuda_executor : base_executor<cuda_executor, Blocking, ProtoAllocator> {
+struct cuda_executor {
   using shape_type = typename std::array<int, 6>;
 
+  template <typename Executor, instance_of_base<cuda_executor, Executor> = 0>
+  friend bool operator==(const cuda_executor& lhs,
+                         const Executor& rhs) noexcept {
+    return std::is_same<cuda_executor, Executor>::value;
+  }
+
+  template <typename Executor, instance_of_base<cuda_executor, Executor> = 0>
+  friend bool operator!=(const cuda_executor& lhs,
+                         const Executor& rhs) noexcept {
+    return !operator==(lhs, rhs);
+  }
```

Comment:
Since this is true of all executors (I might be misremembering, so feel free to correct me), you can "require" all executors to have just this equality if this is outside by `template <class Executor, enable_if_executor<Executor> = 0>`

Reply:
My reasoning behind this is if we need a stateful property for any 1 executor down the line then just comparing type or having a common equality operator will not work.

As mentioned before I think lets first finalize the properties we need then we can go ahead with a different approach.

> executors to have just this equality if this is outside by template <class Executor, enable_if_executor<Executor> = 0>
This will create a circular dependency since `enable_if_executor` relies on checking if the equality operator Is overloaded which in turn will check if `enable_if_executor`.

Reply:
What does the standard say about executors with state and their equality?

Reply:
Its mentioned in [2.2.9](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2020/p0443r12.html). 
For properties `std::query(exec1,p) == std::query(exec2,p)` should be true for all properties. 
Other additional properties, that cannnot be observed via `std::query` are left to implementer to decide if they want to add it to equality check or not. So state information will be left to the implementer.